apps/ewallet/lib/ewallet/gates/reset_password_gate.ex

```diff
@@ -0,0 +1,69 @@
+defmodule EWallet.ResetPasswordGate do
+  @moduledoc """
+  Handles a user's password reset.
+  """
+  alias EWalletDB.{ForgetPasswordRequest, User}
+
+  @doc """
+  Creates a reset password reset.
+  """
+  @spec request(map()) :: %ForgetPasswordRequest{}
+  def request(email) do
+    with {:ok, user} <- get_user_by_email(email),
+         {_, _} <- ForgetPasswordRequest.disable_all_for(user),
+         {:ok, request} <- ForgetPasswordRequest.generate(user) do
+      {:ok, request}
+    else
+      error -> error
+    end
+  end
+
+  @doc """
+  Verifies a reset password request and updates the password.
+  """
+  @spec update(String.t(), String.t(), String.t(), String.t()) :: {:ok, %User{}}
+  def update(email, token, password, password_confirmation) do
+    with {:ok, user} <- get_user_by_email(email),
+         {:ok, request} <- get_request(user, token),
+         {:ok, user} <- update_password(request, password, password_confirmation),
+         {_, _} <- ForgetPasswordRequest.disable_all_for(user) do
+      {:ok, user}
+    else
+      error -> error
+    end
+  end
+
+  # Private functions
+
+  defp get_user_by_email(nil), do: {:error, :user_email_not_found}
+
+  defp get_user_by_email(email) do
+    case User.get_by_email(email) do
+      nil -> {:error, :user_email_not_found}
```

Comment:
I don't think we should return that error when requesting a reset. This is letting know a potential attacker that an email isn't used in the system. When he actually finds an email used, he won't receive any error and will have been able to guess private information. I think we should always return `:ok` when a reset is done with a message like `A reset email has been sent to the given address if it was found in our system.`. Thoughts?